src/Kmd.Logic.FileSecurity.Client/FileSecurityClient.cs

```diff
@@ -61,6 +61,50 @@ public async Task<CreateCertificateResponse> CreateCertificate(CreateCertificate
             }
         }
 
+        /// <summary>
+        /// Get certificate
+        /// </summary>
+        /// <param name="certificateId"></param>
+        /// <returns></returns>
+        public async Task<CertificateResponse> GetCertificate(Guid certificateId)
+        {
+            var client = this.CreateClient();
+
+            using (var certificateDetailsResponse = await client.GetCertificatesWithHttpMessagesAsync(
+                 this._options.SubscriptionId,
+                 certificateId).ConfigureAwait(false))
+            {
+                switch (certificateDetailsResponse?.Response?.StatusCode)
+                {
+                    case System.Net.HttpStatusCode.OK:
+                        return certificateDetailsResponse.Body;
+
+                    case System.Net.HttpStatusCode.NotFound:
+                        return null;
```

Comment:
I would throw `FileSecurityException` here too. With message that we could not find certificate with id {certificateId}